.github/workflows/cd.yml

```diff
@@ -0,0 +1,39 @@
+name: Continuous deployment
+
+run-name: "CI run '${{ github.head_ref || github.ref_name }}'"
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  deploy:
+    name: Check format
+    runs-on: ubuntu-latest
+    steps:
+        - name: Install stable toolchain
+          uses: dtolnay/rust-toolchain@stable
+          with:
+            toolchain: stable
+  
+        - name: Checkout sources
+          uses: actions/checkout@v4
+          with:
+            ref: ${{ inputs.git-ref || github.ref_name }}
+  
+        - name: Publish
+          env:
+            CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+            GIT_TOKEN: ${{ secrets.RELEASE_PLZ_TOKEN }}
+          run: |
+            # Install release-plz
+            cargo install --force --locked --version 0.3.30 release-plz
+  
+            # Bump the version
+            /home/runner/.cargo/bin/release-plz release --config .config/release-plz.toml
```

Comment:
do we expect to run this PR on every merge to main?

Reply:
Yes. `release-plz release` is safe to run on main. It was designed that way. It doesn't do anything if the version hasn't changed. It only picks up new, unpublished versions.

Reply:
Having said that, if there is a way to only run it on version bump commits that would be even better. It's also why I think it would be better to publish the crate itself straight from the beta workflow. If we do that, we won't need this `cd` workflow at all.

Reply:
we could potentially use Commit checks yeah, but it's a bit more involved. If it does nothing unless we bump the version, let's leave it as it is for now 👍